### ISSUE TITLE
Why I need to bind(this) in jsx event handler

### ISSUE THREAD

**Issue by snowmagic1** [survivejs/webpack_react#311]
Hi,

So this is more like a question than a real issue.

When I read chapter 04 I found I have to explicitly bind(this) to the event handler(see code below), otherwise 'this' in handleClick will be 'undefined'.

I understand this is probably due to handleClick is called as a global function plus es6 method by default is in strict mode.

What I don't understand is that if I use es5 syntax like React.createClass, why I don't need to do this binding and this in handleClick is just working fine.

I also expect React will do this binding for me.

```javascript
export default class App extends React.Component {  
    handleClick(){
        this.setState({ .....
   }

render() {
    return (
        <div>
            <button className="add-note" onClick={this.handleClick.bind(this)} />
            <Notes notes={this.state.notes} />
        </div>
    );
  }
}
```

**Comment by bebraw** [comment]
Autobinding is a feature of `createClass`. This was explicitly removed from the class based definitions as discussed at [React release notes](https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#autobinding).

I know it's a little painful. Earlier on I used to bind at the constructor, but given it's so verbose and difficult to manage, I moved to property initializers. This solution is easier to follow, and more importantly, it works with Flow based typing (included in the next release).

I hope this clears it up. :+1:
